hydradx/model/amm/stableswap_amm.py

```diff
@@ -686,6 +688,15 @@ def remove_uniform(
             agent.holdings[tkn] += delta_tkns[tkn]  # agent is receiving funds, because delta_tkn is a negative number
         return self
 
+    def cash_out(self, agent: Agent, prices: dict[str: float]) -> float:
+        if self.unique_id not in agent.holdings:
+            print(f'Error: agent does not have any shares in {self.unique_id}.')
+            return 0
+
+        share_fraction = agent.holdings[self.unique_id] / self.shares
+        delta_tkns = {tkn: share_fraction * self.liquidity[tkn] for tkn in self.asset_list}  # delta_tkn is positive
```

Comment:
Is there a reason we aren't using the remove_uniform function here? Wouldn't it be more reliable than repeating the calculation? I don't recall if stableswap has any fees on uniform withdrawals for example (maybe it doesn't), but it'd be nice to not have to worry about that?

Reply:
The reason is that remove_uniform modifies the state, whereas here I think we don't want to. I do see your point about consistency, though.

Reply:
What I meant was use simulate_remove_uniform. Can we use that instead?